app/scripts/modules/core/src/pipeline/config/stages/gremlin/GremlinExecutionDetails.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+
+import { ExecutionDetailsSection, IExecutionDetailsSectionProps, StageFailureMessage } from 'core/pipeline';
+
+export function GremlinExecutionDetails(props: IExecutionDetailsSectionProps) {
+  const { stage } = props;
+
+  return (
+    <ExecutionDetailsSection name={props.name} current={props.current}>
+      <div className="row">
+        <div className="col-md-12">
+          <h5>Gremlin Stage Configuration</h5>
+          <dl className="dl-narrow dl-horizontal">
+            {stage.context.gremlinCommandTemplateId && <dt>Command Template ID</dt>}
+            {stage.context.gremlinCommandTemplateId && <dd>{stage.context.gremlinCommandTemplateId}</dd>}
+          </dl>
+          <dl className="dl-narrow dl-horizontal">
+            {stage.context.gremlinTargetTemplateId && <dt>Target Template ID</dt>}
+            {stage.context.gremlinTargetTemplateId && <dd>{stage.context.gremlinTargetTemplateId}</dd>}
+          </dl>
+          <dl className="dl-narrow dl-horizontal">
+            {stage.context.gremlinApiKey && <dt>API Key</dt>}
+            {stage.context.gremlinApiKey && <dd>{stage.context.gremlinApiKey}</dd>}
```

Comment:
minor nit: you could destructure out the fields you need from the context above:
```javascript
const { gremlinCommandTemplateId, gremlinTargetTemplateId, gremlinApiKey } = stage.context;
```
then reference those variables here. Would be a little more readable.